api/src/org/apache/cloudstack/api/response/ZoneResponse.java

```diff
@@ -233,6 +234,9 @@ public void addTag(ResourceTagResponse tag) {
     }
 
     public void setResourceDetails(Map<String, String> details) {
-        this.resourceDetails = details;
+        if (details == null) {
```

Comment:
yes, if its null no point in setting it as nothing will be marshaled by gson/xml processor, also we don't want to consume the map but create a new copy out of it to avoid polluting user's datastructures